packtools/validation_report.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import json
+import logging
+
+from datetime import datetime, date
+
+from packtools.sps.utils import xml_utils
+from tests.sps.validation.test_article_authors import credit_terms_and_urls
+
+
+from packtools.sps.validation import (
+    aff,
+    article_authors,
+    article_license,
+    article_toc_sections,
+    article_xref,
+    dates,
+    front_articlemeta_issue
+)        
+
+LOGGER = logging.getLogger(__name__)
+
+
+def all_validation_classes():
+    """
+    Retorna uma lista de todas as classes de validação disponíveis.
+    """
+    classes = [
+        aff.AffiliationValidation,
+        article_authors.ArticleAuthorsValidation, 
+        article_license.ArticleLicenseValidation, 
+        article_toc_sections.ArticleTocSectionsValidation, 
+        article_xref.ArticleXrefValidation,
+        dates.ArticleDatesValidation, 
+        front_articlemeta_issue.IssueValidation, 
+    ]
+    return classes
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    """
+    Classe que serializa objetos `date` em formato ISO para JSON.
+    """
+    def default(self, obj):
+        if isinstance(obj, date):
+            return obj.isoformat()
+        return super().default(obj)
+
+
+class ValidationReportXML:
+    def __init__(self, file_path, data_file_path=None):
+        self.file_path = file_path
+        self.data_validation = self.get_data_validation(data_file_path)
+
+    def validation_report(self):
+        """
+        Executa todas as classes de validação com as opções fornecidas.
+        Retorna um gerador que produz uma sequência de dicionários que contêm os resultados de cada classe de validação.
+        """
+        xmltree = self.get_xml_tree()
+        for validation_class in all_validation_classes():
+            validation = validation_class(xmltree)
+            result = validation.validate(self.data_validation)
+            yield result
+
+    def save_json(self, output_file_path=None):
+        validation_output = self.validation_report()
+        
+        data = [json.dumps(result) for result in list(validation_output)]
+
+        if output_file_path:                    
+            with open(f"{output_file_path}" , "w") as f:
+                f.write(",\n".join(data))
```

Comment:
você poderia trocar isso incluindo um for  com tratamento de exceção de modo que ficará evidente onde está o problema de codificação para JSON